File: app/src/com/example/portalframe/Slide.java
package com.example.portalframe;

/**
 * One slideshow item: an image id (asset path or URL), an optional caption
 * override (e.g. an "On this day" badge — normal captions are derived at display
 * time from {@link #timeMs}), the capture instant in epoch millis (timezone-
 * adjusted to the photo's local wall clock, or {@link #NO_DATE} when unknown),
 * and whether the photo is portrait (taller than wide) for side-by-side pairing.
 */
public class Slide {
    /** Sentinel for "no capture date available" (e.g. bundled sample slides). */
    public static final long NO_DATE = Long.MIN_VALUE;

    public final String id;
    public final String caption; // caption override, or null to derive from timeMs
    public final long timeMs;    // capture instant (tz-adjusted) or NO_DATE
    public final boolean portrait;

    public Slide(String id, String caption) {
        this(id, caption, NO_DATE, false);
    }

    public Slide(String id, String caption, long timeMs) {
        this(id, caption, timeMs, false);
    }

    public Slide(String id, String caption, long timeMs, boolean portrait) {
        this.id = id;
        this.caption = caption;
        this.timeMs = timeMs;
        this.portrait = portrait;
    }
}

File: app/src/com/portalhacks/frame/Slide.kt
package com.portalhacks.frame

/**
 * One slideshow item: an image id (asset path or URL), an optional caption
 * override (e.g. an "On this day" badge — normal captions are derived at display
 * time from [timeMs]), the capture instant in epoch millis (timezone-adjusted to
 * the photo's local wall clock, or [NO_DATE] when unknown), and whether the photo
 * is portrait (taller than wide) for side-by-side pairing.
 */
class Slide @JvmOverloads constructor(
    @JvmField val id: String,
    @JvmField val caption: String?, // caption override, or null to derive from timeMs
    @JvmField val timeMs: Long = NO_DATE, // capture instant (tz-adjusted) or NO_DATE
    @JvmField val portrait: Boolean = false,
    @JvmField val location: String? = null,
) {
    companion object {
        /** Sentinel for "no capture date available" (e.g. bundled sample slides). */
        const val NO_DATE: Long = Long.MIN_VALUE
    }
}
